File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/C/s936984238_comments_removed.c
#include <stdio.h>

int number;
int score[1000];

int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int read() {
    scanf("%d", &number);
    if (number == 0)
        return 0;
    
    for (int i = 0; i < number; i++) {
        scanf("%d", &score[i]);
    }
    return 1;
}

void solve() {
    qsort(score, number, sizeof(int), compare);
    int sum = 0;
    for (int i = 1; i < number - 1; i++) {
        sum += score[i];
    }
    printf("%d\n", sum / (number - 2));
}

int main() {
    while (read()) {
        solve();
    }
    // End of Code
}

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/Java/Go/s936984238_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

var sc = bufio.NewScanner(os.Stdin)

func main() {
	for read() {
		solve()
	}
}

func read() bool {
	sc.Scan()
	number, _ := strconv.Atoi(sc.Text())
	if number == 0 {
		return false
	}

	score := make([]int, number)
	for i := 0; i < number; i++ {
		sc.Scan()
		score[i], _ = strconv.Atoi(sc.Text())
	}
	sort.Ints(score)
	// Pass the score to solve function
	// We need to pass the score to the solve function
	// We can do this by modifying the solve function to accept score as a parameter
	// and then call it with the score array.
	return true
}

func solve(score []int) {
	sum := 0
	for i := 1; i < len(score)-1; i++ {
		sum += score[i]
	}
	fmt.Println(sum / (len(score) - 2))
}

// End of Code
